AxonDeepSeg/morphometrics/launch_morphometrics_computation.py

```diff
@@ -148,12 +149,44 @@ def remove_outside_nerve(pred_axon, pred_myelin, pred_nerve):
 
     return pred_axon, pred_myelin
 
+def compute_fascicle_axon_density(axon_df, nerve_data):
+    """
+    Uses KMeans clustering to assign axons to fascicles and compute axon density.
+    """
+    fascicle_densities = {}
+    total_axons = len(axon_df)  
+
+    # get fascicle centroids 
+    fascicle_areas = {idx: fascicle["value"] for idx, fascicle in nerve_data["fascicle_areas"].items()}
+    
+    # extract (x, y) coordinates of axons
+    axon_positions = axon_df[["x0 (px)", "y0 (px)"]].values
+
+    # use KMeans clustering to assign axons to fascicles
+    kmeans = KMeans(n_clusters=len(fascicle_areas), random_state=42, n_init=10)
+    axon_df["fascicle_id"] = kmeans.fit_predict(axon_positions)
```

Comment:
ok first of all, this was a very nice idea to solve this problem. However, I think this might introduce some complexity that we don't want to deal with. I know k-means is pretty reliable, but there are some edge cases where this might fail.

There is a simpler way to do this. When you call `get_axon_morphometrics()`, internally there is a watershed segmentation that is performed to separate all objects. This is how morphometrics are computed for every single object in a mask. Without going too much into details, you should use the same function that is used here: https://github.com/axondeepseg/axondeepseg/blob/9e92a0c114e19b830bcd4e10948e7a53e81b35a4/AxonDeepSeg/morphometrics/compute_morphometrics.py#L120

Reply:
(axon_objects contains the actual regions of every individual object, separated)

Reply:
based on you can filter axons like you did previously